src/backend/core/api/custom_metrics_exporter.py

```diff
@@ -0,0 +1,198 @@
+            oldest=Min("created_at"),
+            newest=Max("created_at"),
+        )
+
+        # Prepare user distribution data
+        user_doc_counts = models.DocumentAccess.objects.values("user_id").annotate(
+            doc_count=Count("document_id"), admin_email=F("user__admin_email")
+        )
+
+        # Collect all metrics in one list
+        metrics = []
+
+        # -- User metrics
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("total_users"),
+                "Total number of users",
+                value=user_metrics["total_users"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_users_today"),
+                "Number of active users today",
+                value=user_metrics["active_users_today"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_users_7_days"),
+                "Number of active users in the last 7 days",
+                value=user_metrics["active_users_7_days"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_users_30_days"),
+                "Number of active users in the last 30 days",
+                value=user_metrics["active_users_30_days"],
+            )
+        )
+
+        # -- Document metrics
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("total_documents"),
+                "Total number of documents",
+                value=doc_metrics["total_documents"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("shared_documents"),
+                "Number of shared documents",
+                value=doc_metrics["shared_docs_count"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_documents_today"),
+                "Number of active documents today",
+                value=doc_metrics["active_docs_today"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_documents_7_days"),
+                "Number of active documents in the last 7 days",
+                value=doc_metrics["active_docs_last_7_days"],
+            )
+        )
+        metrics.append(
+            GaugeMetricFamily(
+                prefixed_metric_name("active_documents_30_days"),
+                "Number of active documents in the last 30 days",
+                value=doc_metrics["active_docs_last_30_days"],
+            )
+        )
+
+        # -- Document oldest/newest timestamps
+        if doc_ages["oldest"]:
+            metrics.append(
+                GaugeMetricFamily(
+                    prefixed_metric_name("oldest_document_date"),
```

Comment:
It's recommended to include "timestamp"in the metric name and have a _seconds suffix (from https://prometheus.io/docs/practices/naming/#metric-names)
```suggestion
                    prefixed_metric_name("oldest_document_timestamp_seconds"),
```